src/classes/atailentry.cpp

```diff
@@ -17,36 +17,32 @@
  */
 #include "atailentry.h"
 
-namespace experimental {
-
 ATailEntry::ATailEntry()
     : AEntry::AEntry(DEFAULT_TAIL_POSITION)
 {}
 
 ATailEntry::ATailEntry(int row_id)
-    : AEntry::AEntry(DataPosition(QLatin1String("tails"), row_id))
+    : AEntry::AEntry(DataPosition(DB_TABLE_TAILS, row_id))
 {}
 
-ATailEntry::ATailEntry(TableData table_data)
+ATailEntry::ATailEntry(RowData table_data)
     : AEntry::AEntry(DEFAULT_TAIL_POSITION, table_data)
 {}
 
 const QString ATailEntry::registration()
 {
-    return getData().value(QLatin1String("registration")).toString();
+    return getData().value(DB_TAILS_REGISTRATION).toString();
 }
 
 const QString ATailEntry::type()
 {
     QString type_string;
-    if (!getData().value(QLatin1String("make")).toString().isEmpty())
-        type_string.append(getData().value(QLatin1String("make")).toString() + ' ');
-    if (!getData().value(QLatin1String("model")).toString().isEmpty())
-        type_string.append(getData().value(QLatin1String("model")).toString());
-    if (!getData().value(QLatin1String("variant")).toString().isEmpty())
-        type_string.append('-' + getData().value(QLatin1String("variant")).toString() + ' ');
+    if (!getData().value(DB_TAILS_MAKE).toString().isEmpty())
+        type_string.append(getData().value(DB_TAILS_MAKE).toString() + ' ');
+    if (!getData().value(DB_TAILS_MODEL).toString().isEmpty())
+        type_string.append(getData().value(DB_TAILS_MODEL).toString());
+    if (!getData().value(DB_TAILS_VARIANT).toString().isEmpty())
+        type_string.append('-' + getData().value(DB_TAILS_VARIANT).toString() + ' ');
```

Comment:
This doesnt seem like clean code.
Because you are in the scope of the Tail entry there is no need to call the getter, the class handles its private data without a getter. 
Also see my other comment about RowData being a variant. 
Tl;dr RowData should be QString since everything read is a string anyway. If a class needs to convert a "time string" into ISO class thats its problem. The basic data type from the DB is QString not QVariant. Unless im missing something

Reply:
removed the getter and accessed data directly, I guess that was a leftover from when tableData was protected.

See comment above for QVariant vs QString

Reply:
even if it was private the class can access any data it encapsulates.